lib-find-meal/src/main/java/com/kinandcarta/lib/find/meal/view/FindMealFragment.kt

```diff
@@ -1,68 +1,37 @@
 package com.kinandcarta.lib.find.meal.view
 
+import android.annotation.SuppressLint
 import android.content.Context
-import android.location.Location
-import androidx.lifecycle.ViewModelProvider
 import android.os.Bundle
-import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.FrameLayout
-import android.widget.Toast
-import androidx.core.content.ContextCompat
 import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
-import com.kcc.kmmhackathon.shared.MealsSDK
 import com.kinandcarta.lib.find.meal.R
 import com.kinandcarta.lib.find.meal.adapter.MealsAdapter
-import com.kinandcarta.lib.find.meal.viewmodel.FindMealViewModel
-import kotlinx.coroutines.MainScope
-import kotlinx.coroutines.launch
-import android.Manifest
-import android.app.Activity
-import android.content.Intent
-import android.content.IntentSender
-import android.content.pm.PackageManager
-import android.util.Log
-import com.google.android.gms.common.api.ResolvableApiException
-import com.google.android.gms.location.*
-import com.kcc.kmmhackathon.shared.utility.DistanceUnit
+import com.kinandcarta.lib.find.meal.extension.requestFineLocationPermission
+import com.kinandcarta.lib.find.meal.extension.showToast
 import com.kinandcarta.lib.find.meal.utility.PermissionResultParser
-import kotlinx.coroutines.cancel
+import com.kinandcarta.lib.find.meal.viewmodel.FindMealViewModel
+import dagger.hilt.android.AndroidEntryPoint
 
+@AndroidEntryPoint
 class FindMealFragment : Fragment() {
```

Comment:
I've moved quite a lot out of this file and into the VM, the adapter and extensions to try and keep it as neat as possible. The result is that the view (the fragment) is mostly just responsible for handling interactions and reacting to different states that the view model is in